src/object.rs

```diff
@@ -702,7 +703,7 @@ impl<T: IsA<Object> + SetValue> ObjectExt for T {
             if handler == 0 {
                 Err(BoolError("Failed to connect to signal"))
             } else {
-                Ok(handler as u64)
+                Ok(from_glib(handler))
```

Comment:
`handler as u64`?

Reply:
I meant: you lost conversion.

Reply:
No, it's internally all based on c_ulong now again, like the FFI functions. No cast is needed here unless you meant something else.

Reply:
Seems this not fully so:
```
error[E0277]: the trait bound `signal::SignalHandlerId: translate::FromGlib<u64>` is not satisfied
  --> src\signal.rs:57:5
   |
57 |     from_glib(handle)
   |     ^^^^^^^^^ the trait `translate::FromGlib<u64>` is not implemented for `signal::SignalHandlerId`
   |
   = help: the following implementations were found:
             <signal::SignalHandlerId as translate::FromGlib<u32>>
   = note: required by `translate::from_glib`
```

Reply:
There was a cast right in the line above, which would have to be removed. Done! Thanks for noticing

Reply:
FWIW, this would've only failed on Windows (32/64 bit) and 32 bit variants of other platforms. Elsewhere c_ulong and u64 are equivalent.